src/unit_tests/agentlessd/test_agentlessd.c

```diff
@@ -0,0 +1,205 @@
+/*
+ * Copyright (C) 2015, Wazuh Inc.
+ *
+ * This program is free software; you can redistribute it
+ * and/or modify it under the terms of the GNU General Public
+ * License (version 2) as published by the FSF - Free Software
+ * Foundation.
+ */
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <setjmp.h>
+#include <cmocka.h>
+#include <stdio.h>
+
+#include "headers/shared.h"
+#include "agentlessd/agentlessd.h"
+#include "../wrappers/common.h"
+#include "../wrappers/wazuh/shared/debug_op_wrappers.h"
+#include "../wrappers/wazuh/shared/mq_op_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+#include "../wrappers/posix/unistd_wrappers.h"
+#include "../wrappers/libc/stdio_wrappers.h"
+
+// internal use (static) not in agentlessd/agentlessd.h
+int save_agentless_entry(const char *host, const char *script, const char *agttype);
+int send_intcheck_msg(const char *script, const char *host, const char *msg);
+int send_log_msg(const char *script, const char *host, const char *msg);
+int gen_diff_alert(const char *host, const char *script, time_t alert_diff_time);
+#define BUFFER_SIZE OS_MAXSTR - (OS_LOG_HEADER * 2)
+#define STR_MORE_CHANGES "More changes..."
+
+static int test_setup(void **state) {
+    (void) state;
+    test_mode = 1;
+    return OS_SUCCESS;
+}
+
+static int test_teardown(void **state) {
+    (void) state;
+    test_mode = 0;
+    return OS_SUCCESS;
+}
+
+#define HOST "host"
+#define SCRIPT "script"
+#define AGTTYPE "agttype"
+#define MSG "msg"
+#define STR_HELPER(s) #s
+#define STR(s) STR_HELPER(s)
+#define ADT 1667392718
```

Comment:
Complexity nitpick: defines/includes/global variables in an ordered way will improve readability!